app/services/submission/plant_trial_finalizer.rb

```diff
@@ -35,10 +35,80 @@ def create_new_trait_descriptors
 
   def create_scoring
```

Comment:
Well, normally I would never be able to write so long a method in a single stroke. This one grew from addition to addition. Applied naive refactoring ;).